### DISCUSSION TITLE
Pie Charts

### DISCUSSION THREAD

**Discussion by thorinas** [Budibase/budibase#13953] [Help] [answered]
Hi, 

I wonder if anyone can help I'm trying to make a pie chart that shows the percentage of vehicles that have a cretin tag ie, vor, in service, mot, ect.

Iv made the field an option field where they just chose the correct tag and created a pie chart but its not displaying the data correctly, do i need to add  code to the block?  
![image](https://github.com/Budibase/budibase/assets/71787922/ab122ccf-0e40-46de-8ebb-3a45f434b5c8)
![image](https://github.com/Budibase/budibase/assets/71787922/1fe543b7-c434-4ed2-8d5b-d1cb02d43a14)

**Comment by ConorWebb96** [comment]
Hey @thorinas,

You likely will need change the option you've selected for the data column. This needs a number for it to work. Your best bet would be including these numbers somewhere within your table row, so these can then be used to populate your pie chart.

Your field may be called `percentage_of_vehicles` with the number contained within that will be used to populate the pie chart.

I hope this helps.

**Reply by thorinas** [reply]
Sorry how would that work. As all I need it to do is to count the number of times that word is found.

**Reply by ConorWebb96** [reply, marked as answer]
Sounds like you need to use [custom queries](https://docs.budibase.com/docs/data-sources#custom-queries).

You may be able to do grouping using the [rest api](https://docs.budibase.com/docs/rest) along with the [public api](https://docs.budibase.com/reference/rowsearch). As you are able to write JS that could do the grouping of data for you within the transformer.

**Comment by phamalda** [comment]
Good day, all. I just created a new [thread](https://github.com/Budibase/budibase/discussions/16626) around creating a pie chart. @ConorWebb96, I'm really curious if you have any documented steps in employing those [rest api](https://docs.budibase.com/docs/rest) and [public api](https://docs.budibase.com/reference/rowsearch) steps. Not sure if that's possible with an existing Budibase table. Thanks a ton for everyone's time and effort.
